File: app/src/main/java/com/example/myapplication2/listAdapter.java
package com.example.myapplication2;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.ArrayList;

public class listAdapter extends RecyclerView.Adapter<listAdapter.ListViewHolder> {
    private ArrayList<tense> listtense;
    private OnItemClickCallback onItemClickCallback;

    public listAdapter(ArrayList<tense> listtense) {
        this.listtense = listtense;
    }

    public void setOnItemClickCallback(OnItemClickCallback onItemClickCallback) {
        this.onItemClickCallback = onItemClickCallback;
    }

    @NonNull
    @Override
    public ListViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_material, parent, false);
        return new ListViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ListViewHolder holder, int position) {
        tense tense = listtense.get(position);
        holder.imgPhoto.setImageResource(tense.getFoto());
        holder.tvName.setText(tense.getName());

        holder.itemView.setOnClickListener(v -> {
            if (onItemClickCallback != null) {
                onItemClickCallback.onItemClicked(listtense.get(holder.getAdapterPosition()));
            }
        });
    }

    @Override
    public int getItemCount() {
        return listtense.size();
    }

    public static class ListViewHolder extends RecyclerView.ViewHolder {
        ImageView imgPhoto;
        TextView tvName;

        public ListViewHolder(@NonNull View itemView) {
            super(itemView);
            imgPhoto = itemView.findViewById(R.id.iv_material);
            tvName = itemView.findViewById(R.id.tv_title_material);
        }
    }

    public interface OnItemClickCallback {
        void onItemClicked(tense data);
    }
}

File: app/src/main/java/com/example/myapplication2/listAdapter.kt
package com.example.myapplication2

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide

class listAdapter(private val listTense: ArrayList<tense>) : RecyclerView.Adapter<listAdapter.ListViewHolder>() {
    private lateinit var onItemClickCallback: OnItemClickCallback

    fun setOnItemClickCallback(onItemClickCallback: OnItemClickCallback) {
        this.onItemClickCallback = onItemClickCallback
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListViewHolder {
        val view: View = LayoutInflater.from(parent.context).inflate(R.layout.item_material, parent, false)
        return ListViewHolder(view)
    }

    override fun onBindViewHolder(holder: ListViewHolder, position: Int) {
        val (name, foto) = listTense[position]

        holder.imgPhoto.setImageResource(foto)
        holder.tvName.text = name

        holder.itemView.setOnClickListener {
            onItemClickCallback.onItemClicked(listTense[holder.adapterPosition])
        }

    }

    override fun getItemCount(): Int = listTense.size

    class ListViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val imgPhoto: ImageView = itemView.findViewById(R.id.iv_material)
        val tvName: TextView = itemView.findViewById(R.id.tv_title_material)

    }

    interface OnItemClickCallback {
        fun onItemClicked(data: tense)
    }
}